File: src/main/java/yv/tils/dc/discord/EmbedManager/whitelist/discord/ForceAdd.java
package yv.tils.dc.discord.EmbedManager.whitelist.discord;

import net.dv8tion.jda.api.EmbedBuilder;
import yv.tils.dc.utils.language.LanguageFile;
import yv.tils.dc.utils.language.LanguageMessage;

import java.awt.*;
import java.util.ArrayList;

public class ForceAdd {
    EmbedBuilder builder = new EmbedBuilder();
    String url = "https://yvnetwork.de/wp-content/uploads/2022/03/YVtils-SMP.png";

    public EmbedBuilder Embed(String mc, String dc) {
        java.util.List<String> list1 = new ArrayList();
        java.util.List<String> list2 = new ArrayList();
        list1.add("MCNAME");
        list2.add(mc);
        list1.add("DCNAME");
        list2.add(dc);
        return builder
                .setTitle(LanguageFile.getMessage(LanguageMessage.EMBED_CMD_WHITELIST_ADD_TITLE))
                .setDescription(new LanguageFile().ListReplacer(LanguageFile.getMessage(LanguageMessage.EMBED_CMD_WHITELIST_ADD_DESC), list1, list2))
                .setColor(new Color(0x85BA4C))
                .setFooter("YVtils-SMP • https://yvnetwork.de/yvtils-smp/spigot", "https://yvnetwork.de/wp-content/uploads/2022/03/YVtils-SMP.png")
                .setAuthor("Whitelist Administration", null, url);
    }

    public EmbedBuilder Replace(String dc, String mc_old, String mc_new) {
        java.util.List<String> list1 = new ArrayList();
        java.util.List<String> list2 = new ArrayList();
        list1.add("DCNAME");
        list2.add(dc);
        list1.add("OLDNAME");
        list2.add(mc_old);
        list1.add("NEWNAME");
        list2.add(mc_new);
        return builder
                .setTitle(new LanguageFile().ListReplacer(LanguageFile.getMessage(LanguageMessage.EMBED_CMD_WHITELIST_REPLACE_TITLE), list1, list2))
                .setDescription(new LanguageFile().ListReplacer(LanguageFile.getMessage(LanguageMessage.EMBED_CMD_WHITELIST_REPLACE_DESC), list1, list2))
                .setColor(new Color(7719960))
                .setFooter("YVtils-SMP • https://yvnetwork.de/yvtils-smp/spigot", "https://yvnetwork.de/wp-content/uploads/2022/03/YVtils-SMP.png")
                .setAuthor("Whitelist Administration", null, url);
    }
}

File: src/main/kotlin/yv/tils/dc/mods/discord/embedManager/commands/whitelist/ForceAdd.kt
package yv.tils.dc.mods.discord.embedManager.whitelist.discord

import net.dv8tion.jda.api.EmbedBuilder
import yv.tils.dc.mods.discord.embedManager.EmbedVars
import yv.tils.dc.utils.color.ColorUtils
import yv.tils.dc.utils.configs.language.LangStrings
import yv.tils.dc.utils.configs.language.Language
import yv.tils.dc.utils.internalAPI.Placeholder

class ForceAdd {
    val builder = EmbedBuilder()

    fun embed(mc: String, dc: String): EmbedBuilder {
        return builder
            .setTitle(ColorUtils().convert(Language().getMessage(LangStrings.EMBED_CMD_WHITELIST_ADD_TITLE)))
            .setDescription(
                ColorUtils().convert(
                    Placeholder().replacer(
                        Language().getMessage(LangStrings.EMBED_CMD_WHITELIST_ADD_DESC),
                        listOf("mcName", "dcName"),
                        listOf(mc, dc)
                    )
                )
            )
            .setColor(EmbedVars.successColor)
            .setFooter(EmbedVars.footerText, EmbedVars.footerIcon)
            .setAuthor(EmbedVars.authorName, EmbedVars.authorLink, EmbedVars.authorIcon)
    }

    fun embedReplace(dc: String, mcOld: String, mcNew: String): EmbedBuilder {
        return builder
            .setTitle(
                ColorUtils().convert(
                    Placeholder().replacer(
                        Language().getMessage(LangStrings.EMBED_CMD_WHITELIST_REPLACE_TITLE),
                        listOf("dcName", "oldName", "newName"),
                        listOf(dc, mcOld, mcNew)
                    )
                )
            )
            .setDescription(
                ColorUtils().convert(
                    Placeholder().replacer(
                        Language().getMessage(LangStrings.EMBED_CMD_WHITELIST_REPLACE_DESC),
                        listOf("dcName", "oldName", "newName"),
                        listOf(dc, mcOld, mcNew)
                    )
                )
            )
            .setColor(EmbedVars.infoColor)
            .setFooter(EmbedVars.footerText, EmbedVars.footerIcon)
            .setAuthor(EmbedVars.authorName, EmbedVars.authorLink, EmbedVars.authorIcon)
    }
}